flow/actorcompiler/ActorParser.cs

```diff
@@ -496,6 +538,20 @@ void ParseActorHeading(Actor actor, TokenRange toks)
                     throw new Error(actor.SourceLine, "Unrecognized tokens preceding parameter list in actor declaration");
                 }
             }
+            if (errorMessagePolicy.ActorsNoDiscardByDefault() && !actor.attributes.Contains("[[flow_allow_discard]]")) {
+                if (actor.IsCancellable() && actor.returnType != null)
```

Comment:
Based on my other suggested change, this could become:

```suggestion
                if (actor.IsCancellable())
```